akd/src/directory.rs

```diff
@@ -65,11 +65,14 @@ where
     pub async fn new(storage: StorageManager<S>, vrf: V) -> Result<Self, AkdError> {
         let azks = Directory::<TC, S, V>::get_azks_from_storage(&storage, false).await;
 
-        if azks.is_err() {
-            // generate a new azks if one is not found
-            let azks = Azks::new::<TC, _>(&storage).await?;
-            // store it
-            storage.set(DbRecord::Azks(azks.clone())).await?;
+        if let Err(AkdError::Storage(StorageError::NotFound(_))) = azks {
+            // generate + store a new azks only if one is not found
```

Comment:
Can we log here that AZKS not found (and the error string), and we are creating a new one along with a root?

Reply:
Sure! I am not sure the error string will have much value, but it doesn't hurt to log it!